src/site/apt/sessions/session-access-grants.apt.vm

```diff
@@ -3,6 +3,135 @@ Access Grants
     {{{https://docs.inrupt.com/ess/latest/security/access-requests-grants/}Access Grants}} provide an alternative way to give access to your Solid resources. The Inrupt Java Client Libraries
     provide tooling for working with Access Grants.
 
+    To use {{{https://docs.inrupt.com/ess/latest/security/access-requests-grants/}Access Grants}}, the resources one wants to access must have {{{https://docs.inrupt.com/ess/latest/security/access-requests-grants/#access-grants-acp}ACP that enables the use of access grants}} and the Authorization Service must be able to exchange an access grant for a token (for example such as a {{{https://docs.inrupt.com/ess/latest/services/service-uma/}UMA Service}}).
+
+* ACP that enables the use of Access Grants
+
+    One needs to edit the ACP of the resource the access grant will be used for. 
+    
+** Manually adding triples to ACP
+    
+    This can be done manually through using a resource management interface such as {{{https://podpro.dev/} Pod Pro}}.
+
+    The next example, contains a new access control that matches for <<<http://www.w3.org/ns/solid/vc#SolidAccessGrant>>>> and allows Read and Write for a Solid Container and its contents. 
+
++---
+prefix acp: <http://www.w3.org/ns/solid/acp#>
+prefix acl: <http://www.w3.org/ns/auth/acl#>
+
+# ACR for a Solid Container
+<>
+        a acp:AccessControlResource ;
+        # Add access control for the container itself
+        acp:accessControl <#newAccessControl> ;
+        # Add access control for contained resources
+        acp:memberAccessControl <#newAccessControl> .
+
+<#newAccessControl>
+        a acp:AccessControl ;
+        # The access control applies one policy
+        acp:apply <#newPolicy> .
+
+<#newPolicy>
+        a acp:Policy ;
+        # The policy requires one matcher to be satisfied
+        acp:allOf <#newMatcher> ;
+        # The policy grants read and write when its matcher is satisfied
+        acp:allow acl:Read, acl:Write .
+
+<#newMatcher>
+        a acp:Matcher ;
+        # The matcher is satisfied if a VC of type Solid Access Grant was presented
+        acp:vc <http://www.w3.org/ns/solid/vc#SolidAccessGrant> .
++---
+
+** Adding triples to ACP in code
+
+    One can write the above triples to an ACL auxiliary resource using code. The next code appends to an existing ACP, a new access control that matches for <<<http://www.w3.org/ns/solid/vc#SolidAccessGrant>>> and allows Read and Write over a Solid Resource. We are using only the <<<http://www.w3.org/ns/solid/acp#accessControl>>> predicate in the <<<AccessControlResource>>>, so contained resources are not affected.
+
++---
+import com.inrupt.client.Headers;
+import com.inrupt.client.Request;
+import com.inrupt.client.Response;
+import com.inrupt.client.solid.SolidSyncClient;
+import com.inrupt.client.spi.RDFFactory;
```

Comment:
```suggestion
import com.inrupt.client.api.RDFFactory;
```

I know it is not the current structure, but I'm really surprised by this acronym, I would have expected API instead ... is it a Typo?

[SPI as an acronym](https://en.wikipedia.org/wiki/SPI) doesn't seem to be very commonly used in a way that matches what we potentially intended.

What does it mean here?

Reply:
SPI is exactly what is intended. This is a standard way of supporting this sort of structure in Java.

Reply:
In terms of references, this is very common in JakartaEE APIs, as well as MicroProfile APIs:

* JakartaEE [JSON-Bind](https://github.com/jakartaee/jsonb-api/blob/master/api/src/main/java/jakarta/json/bind/spi/package-info.java)
* JakartaEE [CDI](https://github.com/jakartaee/cdi/blob/master/api/src/main/java/jakarta/enterprise/inject/spi/package-info.java)
* Jakarta [Validation](https://github.com/jakartaee/validation/blob/master/src/main/java/jakarta/validation/spi/package-info.java)
* Eclipse Microprofile [Config](https://github.com/eclipse/microprofile-config/blob/master/api/src/main/java/org/eclipse/microprofile/config/spi/package-info.java)
* Eclipse Micorprofile [Health](https://github.com/eclipse/microprofile-health/blob/master/api/src/main/java/org/eclipse/microprofile/health/spi/package-info.java)
* [Apache Camel](https://github.com/apache/camel/blob/main/core/camel-api/src/main/java/org/apache/camel/spi/package.html)
* Apache [CXF](https://github.com/apache/cxf/tree/main/core/src/main/java/org/apache/cxf/common/spi)
* Apache [Accumulo](https://github.com/apache/accumulo/blob/main/core/src/main/java/org/apache/accumulo/core/spi/package-info.java)

There are many, many more examples, but this _is_ the way java projects structure plugin architectures, and this code follows that model.